openTSNE/affinity.py

```diff
@@ -255,9 +255,17 @@ def check_perplexity(self, perplexity):
 def build_knn_index(
     data, method, k, metric, metric_params=None, n_jobs=1, random_state=None
 ):
+    if not sp.issparse(data) and metric in ["angular", "euclidean", "manhattan", "hamming", "dot"]:
+        approxmethod = nearest_neighbors.Annoy
+    else:
+        approxmethod = nearest_neighbors.NNDescent
+
     methods = {
         "exact": nearest_neighbors.BallTree,
-        "approx": nearest_neighbors.NNDescent,
+        "auto": approxmethod,
```

Comment:
`approxmethod` should be changed to `approx_method`

Reply:
ok done